### ISSUE TITLE
`fetch` method adds unknown characters to the end of the file corrupting it

### ISSUE THREAD

**Issue by sebastianewak** [magic-script/magic-script-polyfills#12]
The method used for saving:
```
fetch(this.writePath, {
      method: 'PUT',
      body: JSON.stringify(json, null, 2)
});
```

**Comment by creationix** [comment]
Fixed in 7e702135c3e9a4c079a8ed22c091ecaf184a1f4c
Published as magic-script-polyfills@2.2.1

**Comment by creationix** [comment]
Also, you don't have to use fetch for file reading and writing, it's just an API wrapper over `fs.readFileStream` and `fs.writeFileStream` that reuses known APIs from the browser.

```js
// Use internal functions directly
import { writeFileStream, readFileStream } from 'magic-script-polyfills/src/fs.js';
import { consume, binToStr } from 'magic-script-polyfills/src/utils.js';

main();
async function main () {
  const filepath = '/tmp/file.json';
  // The value here can be an iterator, an async interator, a string, or a buffer or any combination of them.
  await writeFileStream(filepath, 'Hello World\n');

  // Read the file back in chunks as ArrayBuffer
  const stream = await readFileStream(filepath);
  print('stream', stream);
  const bin = await consume(stream);
  print('bin', bin);
  const str = binToStr(bin);
  print('str', str);
}
```

Technically there are not public APIs yet, but I'll probably still bump the major version if I ever introduce a breaking change in them.  The long-term plan is probably to move these to another library since they are not polyfills (polyfills needs to be existing standard APIs), but very useful.
